capsules/src/log.rs

```diff
@@ -710,7 +710,7 @@ impl<'a, F: Flash + 'static> LogWrite<'a> for Log<'a, F> {
         if self.state.get() != State::Idle {
             // Log busy, try appending again later.
             return Err((ReturnCode::EBUSY, Some(buffer)));
-        } else if length <= 0 || buffer.len() < length {
+        } else if buffer.len() < length {
```

Comment:
These two don't seem to be the same to me. What if length equals zero?

Reply:
good catch, that's my mistake. It should be ==. Fixed.